url49584c1d41_Arrem_GGS_tgz-pJ8-com_gamezgalaxy_test_console_MainJ8/src/com/gamezgalaxy/GGS/API/EventSystem.java

```diff
@@ -0,0 +1,112 @@
+/*******************************************************************************
+ * Copyright (c) 2012 GamezGalaxy.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the GNU Public License v3.0
+ * which accompanies this distribution, and is available at
+ * http://www.gnu.org/licenses/gpl.html
+ ******************************************************************************/
+package com.gamezgalaxy.GGS.API;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import com.gamezgalaxy.GGS.server.Server;
+
+public class EventSystem {
+	
+	private Server server;
+	
+	public EventSystem(Server server) {
+		this.server = server;
+	}
+	
+	public void callEvent(Event event) {
+		EventList events = event.getEvents();
+		RegisteredListener[] listeners = events.getRegisteredListeners();
+		for (RegisteredListener listen : listeners) {
+			try {
+				listen.execute(event);
+			}
+			catch (Exception e) {
+				server.Log("==!EVENT ERROR!==");
+				e.printStackTrace();
+			}
+		}
+	}
+	public void registerEvents(Listener l) {
+		for (Map.Entry<Class<? extends Event>, Set<RegisteredListener>> entry : addMuffins(l).entrySet()) {
+            try {
+				getEventListeners(getRegistrationClass(entry.getKey())).registerAll(entry.getValue());
+			} catch (IllegalAccessException e) {
+				server.Log("==!EVENT ERROR!==");
+				e.printStackTrace();
+			}
+        }
+	}
+	
+	private EventList getEventListeners(Class<? extends Event> type) {
+        try {
+            Method method = getRegistrationClass(type).getDeclaredMethod("getEventList");
+            method.setAccessible(true);
+            return (EventList) method.invoke(null);
+        } catch (Exception e) {
+        	server.Log("==!EVENT ERROR!==");
+            e.printStackTrace();
+            return null;
+        }
+    }
+	
+	private Class<? extends Event> getRegistrationClass(Class<? extends Event> clazz) throws IllegalAccessException {
+        try {
+            clazz.getDeclaredMethod("getEventList");
+            return clazz;
+        } catch (NoSuchMethodException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)